### PR TITLE
Fix null-ref on bad arg to help

### PR DESCRIPTION
**Pull request by BrennanConroy**
#862

### CODE REVIEW

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs [hunk 1]
```diff
@@ -332,6 +332,10 @@ public void ShowHelp(string commandName = null)
             else
             {
                 target = Commands.SingleOrDefault(cmd => string.Equals(cmd.Name, commandName, StringComparison.OrdinalIgnoreCase));
+                if (target == null)
+                {
+                    throw new Exception(string.Format("TODO: Error: unexpected value '{0}' for option 'help'", commandName));
```
Comment (muratg) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: String interpolation

**Comment by muratg** [comment]
:shipit:

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs (hunk 1 above)
Comment (BrennanConroy) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: Sent from my HTC

----- Reply message -----

Comment (moozzyk) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: Don't throw `Exception`. This should be either `ArgumentException` or `InvalidOperationException`

**Comment by BrennanConroy** [comment]
Updated exception. We can't use c#6 features because when building dnx-mono we use an old .NET compiler version. @davidfowl said there might be a way around this

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs [hunk 2]
```diff
@@ -332,6 +332,10 @@ public void ShowHelp(string commandName = null)
             else
             {
                 target = Commands.SingleOrDefault(cmd => string.Equals(cmd.Name, commandName, StringComparison.OrdinalIgnoreCase));
+                if (target == null)
+                {
+                    throw new ArgumentException(string.Format("TODO: Error: unexpected value '{0}' for option 'help'", commandName));
+                }
```
Comment (moozzyk) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: consider passing the name of the parameter as the second parameter to the `ArgumentException` ctor. (Note, I am not feeling strongly about this - this is command line so probably parameter names don't matter)

**Comment by moozzyk** [comment]
:ship::it:

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs (hunk 2 above)
Comment (davidfowl) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: Get rid of the TODO

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs [hunk 3]
```diff
@@ -332,6 +332,10 @@ public void ShowHelp(string commandName = null)
             else
             {
                 target = Commands.SingleOrDefault(cmd => string.Equals(cmd.Name, commandName, StringComparison.OrdinalIgnoreCase));
+                if (target == null)
+                {
+                    throw new ArgumentException(string.Format("Error: unexpected value '{0}' for option 'help'", commandName), "commandName");
+                }
```
Comment (ChengTian) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: If we run into this case, it means that the user is trying to do something like
```
dnu help invalid_subcommand
```
We may want to make the message more friendly:
```
$"'{commandName}' is not a subcommand of '{cmd.Parent.Name}'"
```

**Pull request closed without merging**